fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -555,7 +564,9 @@ class ReadIteratorPool {
 	std::mutex mutex;
 	// incrementing counter for every new iterator creation, to uniquely identify the iterator in returnIterator().
 	uint64_t index;
+	uint64_t deletedUptoIndex;
 	uint64_t iteratorsReuseCount;
+	PromiseStream<Void> deleteIteratorsPromise;
```

Comment:
I think `ThreadReturnPromiseStream` should be used here. `PromiseStream` is not thread safe.